[2025-07-26T13:25:37.746+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-07-26T13:25:37.977+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: sample_ETL.load_order_and_delivery_data manual__2025-07-26T13:25:33.358879+00:00 [queued]>
[2025-07-26T13:25:37.990+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: sample_ETL.load_order_and_delivery_data manual__2025-07-26T13:25:33.358879+00:00 [queued]>
[2025-07-26T13:25:37.991+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 1
[2025-07-26T13:25:38.026+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): load_order_and_delivery_data> on 2025-07-26 13:25:33.358879+00:00
[2025-07-26T13:25:38.035+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=288) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-07-26T13:25:38.037+0000] {standard_task_runner.py:63} INFO - Started process 292 to run task
[2025-07-26T13:25:38.036+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'sample_ETL', 'load_order_and_delivery_data', 'manual__2025-07-26T13:25:33.358879+00:00', '--job-id', '91', '--raw', '--subdir', 'DAGS_FOLDER/sample_ETL.py', '--cfg-path', '/tmp/tmp9oit9f25']
[2025-07-26T13:25:38.039+0000] {standard_task_runner.py:91} INFO - Job 91: Subtask load_order_and_delivery_data
[2025-07-26T13:25:38.111+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/settings.py:195 DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
[2025-07-26T13:25:38.213+0000] {task_command.py:426} INFO - Running <TaskInstance: sample_ETL.load_order_and_delivery_data manual__2025-07-26T13:25:33.358879+00:00 [running]> on host 88b2e53952eb
[2025-07-26T13:25:38.484+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='sample_ETL' AIRFLOW_CTX_TASK_ID='load_order_and_delivery_data' AIRFLOW_CTX_EXECUTION_DATE='2025-07-26T13:25:33.358879+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-07-26T13:25:33.358879+00:00'
[2025-07-26T13:25:38.486+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-07-26T13:25:38.501+0000] {logging_mixin.py:188} INFO - ðŸš€ Starting ETL with JSON files
[2025-07-26T13:25:38.505+0000] {conn.py:396} INFO - <BrokerConnection client_id=kafka-python-producer-1, node_id=bootstrap-0 host=kafka:9092 <connecting> [IPv4 ('172.18.0.7', 9092)]>: connecting to kafka:9092 [('172.18.0.7', 9092) IPv4]
[2025-07-26T13:25:38.512+0000] {conn.py:618} INFO - <BrokerConnection client_id=kafka-python-producer-1, node_id=bootstrap-0 host=kafka:9092 <checking_api_versions_recv> [IPv4 ('172.18.0.7', 9092)]>: Broker version identified as 2.6
[2025-07-26T13:25:38.513+0000] {conn.py:457} INFO - <BrokerConnection client_id=kafka-python-producer-1, node_id=bootstrap-0 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Connection complete.
[2025-07-26T13:25:38.621+0000] {conn.py:396} INFO - <BrokerConnection client_id=kafka-python-producer-1, node_id=1 host=kafka:9092 <connecting> [IPv4 ('172.18.0.7', 9092)]>: connecting to kafka:9092 [('172.18.0.7', 9092) IPv4]
[2025-07-26T13:25:38.623+0000] {conn.py:457} INFO - <BrokerConnection client_id=kafka-python-producer-1, node_id=1 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Connection complete.
[2025-07-26T13:25:38.624+0000] {conn.py:945} INFO - <BrokerConnection client_id=kafka-python-producer-1, node_id=bootstrap-0 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Closing connection. 
[2025-07-26T13:25:38.652+0000] {logging_mixin.py:188} INFO - âœ… Produced 124 records to topic 'orders_topic'
[2025-07-26T13:25:38.653+0000] {logging_mixin.py:188} INFO - âœ… Orders producer completed
[2025-07-26T13:25:38.655+0000] {conn.py:396} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=kafka:9092 <connecting> [IPv4 ('172.18.0.7', 9092)]>: connecting to kafka:9092 [('172.18.0.7', 9092) IPv4]
[2025-07-26T13:25:38.660+0000] {conn.py:618} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=kafka:9092 <checking_api_versions_recv> [IPv4 ('172.18.0.7', 9092)]>: Broker version identified as 2.6
[2025-07-26T13:25:38.661+0000] {conn.py:457} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Connection complete.
[2025-07-26T13:25:38.662+0000] {subscription_state.py:181} INFO - Updating subscribed topics to: ('orders_topic',)
[2025-07-26T13:25:38.709+0000] {conn.py:396} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=1 host=kafka:9092 <connecting> [IPv4 ('172.18.0.7', 9092)]>: connecting to kafka:9092 [('172.18.0.7', 9092) IPv4]
[2025-07-26T13:25:38.710+0000] {conn.py:457} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=1 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Connection complete.
[2025-07-26T13:25:38.711+0000] {conn.py:945} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Closing connection. 
[2025-07-26T13:25:38.815+0000] {cluster.py:401} INFO - Coordinator for group/etl_group_orders is ('coordinator-1', 'kafka', 9092, None)
[2025-07-26T13:25:38.816+0000] {base.py:788} INFO - Discovered coordinator coordinator-1 for group etl_group_orders
[2025-07-26T13:25:38.817+0000] {base.py:848} INFO - Starting new heartbeat thread
[2025-07-26T13:25:38.819+0000] {consumer.py:365} INFO - Revoking previously assigned partitions () for group etl_group_orders
[2025-07-26T13:25:38.819+0000] {base.py:419} INFO - Failed to join group etl_group_orders: NodeNotReadyError: coordinator-1
[2025-07-26T13:25:38.821+0000] {conn.py:396} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=coordinator-1 host=kafka:9092 <connecting> [IPv4 ('172.18.0.7', 9092)]>: connecting to kafka:9092 [('172.18.0.7', 9092) IPv4]
[2025-07-26T13:25:38.822+0000] {conn.py:457} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=coordinator-1 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Connection complete.
[2025-07-26T13:25:38.924+0000] {base.py:532} INFO - (Re-)joining group etl_group_orders
[2025-07-26T13:25:38.935+0000] {base.py:637} INFO - Received member id kafka-python-2.2.15-9fe0a3c4-7d40-43ad-a021-e4b5ce479aae for group etl_group_orders; will retry join-group
[2025-07-26T13:25:38.938+0000] {base.py:419} INFO - Failed to join group etl_group_orders: [Error 79] MemberIdRequiredError
[2025-07-26T13:25:38.940+0000] {base.py:532} INFO - (Re-)joining group etl_group_orders
[2025-07-26T13:25:41.953+0000] {base.py:594} INFO - Successfully joined group etl_group_orders <Generation 5 (member_id: kafka-python-2.2.15-9fe0a3c4-7d40-43ad-a021-e4b5ce479aae, protocol: range)>
[2025-07-26T13:25:41.954+0000] {base.py:596} INFO - Elected group leader -- performing partition assignments using range
[2025-07-26T13:25:41.962+0000] {subscription_state.py:254} INFO - Updated partition assignment: [('orders_topic', 0)]
[2025-07-26T13:25:41.963+0000] {consumer.py:249} INFO - Setting newly assigned partitions (('orders_topic', 0),) for group etl_group_orders
[2025-07-26T13:25:54.068+0000] {logging_mixin.py:188} INFO - âœ… Inserted 124 total records into orders
[2025-07-26T13:25:54.089+0000] {base.py:1052} INFO - Stopping heartbeat thread
[2025-07-26T13:25:54.093+0000] {base.py:891} INFO - Leaving consumer group (etl_group_orders).
[2025-07-26T13:25:54.113+0000] {base.py:906} INFO - LeaveGroup request for group etl_group_orders returned successfully
[2025-07-26T13:25:54.117+0000] {conn.py:945} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=1 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Closing connection. 
[2025-07-26T13:25:54.120+0000] {fetcher.py:786} INFO - Fetch to node 1 failed: Cancelled: <BrokerConnection client_id=kafka-python-2.2.15, node_id=1 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>
[2025-07-26T13:25:54.122+0000] {conn.py:945} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=coordinator-1 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Closing connection. 
[2025-07-26T13:25:54.124+0000] {logging_mixin.py:188} INFO - âœ… Orders consumer completed
[2025-07-26T13:25:59.172+0000] {logging_mixin.py:188} INFO - âœ… Produced 124 records to topic 'delivery_topic'
[2025-07-26T13:25:59.173+0000] {logging_mixin.py:188} INFO - âœ… Delivery producer completed
[2025-07-26T13:25:59.176+0000] {conn.py:396} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=kafka:9092 <connecting> [IPv4 ('172.18.0.7', 9092)]>: connecting to kafka:9092 [('172.18.0.7', 9092) IPv4]
[2025-07-26T13:25:59.185+0000] {conn.py:618} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=kafka:9092 <checking_api_versions_recv> [IPv4 ('172.18.0.7', 9092)]>: Broker version identified as 2.6
[2025-07-26T13:25:59.186+0000] {conn.py:457} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Connection complete.
[2025-07-26T13:25:59.188+0000] {subscription_state.py:181} INFO - Updating subscribed topics to: ('delivery_topic',)
[2025-07-26T13:25:59.274+0000] {conn.py:396} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=1 host=kafka:9092 <connecting> [IPv4 ('172.18.0.7', 9092)]>: connecting to kafka:9092 [('172.18.0.7', 9092) IPv4]
[2025-07-26T13:25:59.276+0000] {conn.py:457} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=1 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Connection complete.
[2025-07-26T13:25:59.277+0000] {conn.py:945} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Closing connection. 
[2025-07-26T13:25:59.381+0000] {cluster.py:401} INFO - Coordinator for group/etl_group_delivery is ('coordinator-1', 'kafka', 9092, None)
[2025-07-26T13:25:59.382+0000] {base.py:788} INFO - Discovered coordinator coordinator-1 for group etl_group_delivery
[2025-07-26T13:25:59.383+0000] {base.py:848} INFO - Starting new heartbeat thread
[2025-07-26T13:25:59.384+0000] {consumer.py:365} INFO - Revoking previously assigned partitions () for group etl_group_delivery
[2025-07-26T13:25:59.385+0000] {base.py:419} INFO - Failed to join group etl_group_delivery: NodeNotReadyError: coordinator-1
[2025-07-26T13:25:59.386+0000] {conn.py:396} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=coordinator-1 host=kafka:9092 <connecting> [IPv4 ('172.18.0.7', 9092)]>: connecting to kafka:9092 [('172.18.0.7', 9092) IPv4]
[2025-07-26T13:25:59.387+0000] {conn.py:457} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=coordinator-1 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Connection complete.
[2025-07-26T13:25:59.489+0000] {base.py:532} INFO - (Re-)joining group etl_group_delivery
[2025-07-26T13:25:59.491+0000] {base.py:637} INFO - Received member id kafka-python-2.2.15-8fbf8156-14de-4835-afff-ddf6a3898082 for group etl_group_delivery; will retry join-group
[2025-07-26T13:25:59.492+0000] {base.py:419} INFO - Failed to join group etl_group_delivery: [Error 79] MemberIdRequiredError
[2025-07-26T13:25:59.493+0000] {base.py:532} INFO - (Re-)joining group etl_group_delivery
[2025-07-26T13:26:02.497+0000] {base.py:594} INFO - Successfully joined group etl_group_delivery <Generation 5 (member_id: kafka-python-2.2.15-8fbf8156-14de-4835-afff-ddf6a3898082, protocol: range)>
[2025-07-26T13:26:02.498+0000] {base.py:596} INFO - Elected group leader -- performing partition assignments using range
[2025-07-26T13:26:02.504+0000] {subscription_state.py:254} INFO - Updated partition assignment: [('delivery_topic', 0)]
[2025-07-26T13:26:02.505+0000] {consumer.py:249} INFO - Setting newly assigned partitions (('delivery_topic', 0),) for group etl_group_delivery
[2025-07-26T13:26:14.624+0000] {logging_mixin.py:188} INFO - âœ… Inserted 124 total records into delivery
[2025-07-26T13:26:14.631+0000] {base.py:1052} INFO - Stopping heartbeat thread
[2025-07-26T13:26:14.632+0000] {base.py:891} INFO - Leaving consumer group (etl_group_delivery).
[2025-07-26T13:26:14.636+0000] {base.py:906} INFO - LeaveGroup request for group etl_group_delivery returned successfully
[2025-07-26T13:26:14.637+0000] {conn.py:945} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=1 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Closing connection. 
[2025-07-26T13:26:14.638+0000] {fetcher.py:786} INFO - Fetch to node 1 failed: Cancelled: <BrokerConnection client_id=kafka-python-2.2.15, node_id=1 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>
[2025-07-26T13:26:14.639+0000] {conn.py:945} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=coordinator-1 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Closing connection. 
[2025-07-26T13:26:14.640+0000] {logging_mixin.py:188} INFO - âœ… Delivery consumer completed
[2025-07-26T13:26:14.641+0000] {logging_mixin.py:188} INFO - âœ… Task2 of ETL completed
[2025-07-26T13:26:14.641+0000] {python.py:237} INFO - Done. Returned value was: None
[2025-07-26T13:26:14.642+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-07-26T13:26:14.687+0000] {taskinstance.py:1206} INFO - Marking task as SUCCESS. dag_id=sample_ETL, task_id=load_order_and_delivery_data, run_id=manual__2025-07-26T13:25:33.358879+00:00, execution_date=20250726T132533, start_date=20250726T132537, end_date=20250726T132614
[2025-07-26T13:26:14.745+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 0
[2025-07-26T13:26:14.808+0000] {taskinstance.py:3498} INFO - 1 downstream tasks scheduled from follow-on schedule check
[2025-07-26T13:26:14.829+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
